rpcs3/Emu/RSX/Overlays/overlay_message.cpp

```diff
@@ -0,0 +1,103 @@
+#include "stdafx.h"
+#include "overlay_message.h"
+
+LOG_CHANNEL(testor);
+
+namespace rsx
+{
+	namespace overlays
+	{
+		message_item::message_item(localized_string_id msg_id)
+		{
+			m_expiration_time = get_system_time() + 5'000'000;
+
+			m_text.set_font("Arial", 16);
+			m_text.set_text(msg_id);
+			m_text.auto_resize();
+			m_text.back_color.a = 0.f;
+		}
+
+		u64 message_item::get_expiration() const
+		{
+			return m_expiration_time;
+		}
+
+		compiled_resource message_item::get_compiled()
+		{
+			if (!m_processed)
+			{
+				return {};
+			}
+
+			return m_text.get_compiled();
+		}
+
+		void message_item::update(usz index, u64 time)
+		{
+			m_processed = true;
+
+			if (m_cur_pos != index)
+			{
+				m_cur_pos = index;
+				m_text.set_pos(10, index * 18);
+			}
+
+			const u64 diff = m_expiration_time - time;
+
+			if (diff < 2'000'000)
+			{
+				m_text.fore_color.a = (diff / 2'000'000.f);
```

Comment:
I think for future compatibility this should use the animation_color_interpolate class

Reply:
Seconded. We have animation classes for overlays that are fairly easy to use.